### PR TITLE
[12.0][MIG] account_check_printing_report_sslm102

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Standard migration.

### CODE REVIEW

**Review by AaronHForgeFlow — CHANGES_REQUESTED** [review]
>  File "/.repo_requirements/odoo/odoo/addons/base/models/qweb.py", line 353, in _compiled_fn
    raise QWebException("Error to render compiling AST", e, path, node and etree.tostring(node[0], encoding='unicode'), name)
odoo.addons.base.models.qweb.QWebException: 'NoneType' object is not callable
Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/addons/base/models/qweb.py", line 346, in _compiled_fn
    return compiled(self, append, new, options, log)
  File "<template>", line 1, in template_account_check_printing_report_sslm102_invoice_data_478
TypeError: 'NoneType' object is not callable

Error to render compiling AST
TypeError: 'NoneType' object is not callable
Template: account_check_printing_report_sslm102.invoice_data
Path: /templates/t/div/strong/span[2]
Node: <span name="payment_date" style="padding-right:100mm;float:right;" t-esc="'Date: {}'.format(_format_date_to_partner_lang(o.payment_date, o.partner_id.id))"/>
                

When printing: http://3393814-288-afe5ce.runbot3.odoo-community.org/web?debug=true#id=2&action=140&active_id=7&model=account.payment&view_type=form&menu_id=82

File: account_check_printing_report_sslm102/views/report_check.xml [hunk 1]
```diff
@@ -0,0 +1,104 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <template id="sslm102">
+         <t t-call="web.html_container">
+             <div class="header"/>
+             <div class="article">
+                <t t-foreach="docs" t-as="o">
+                    <div class="page">
+                        <div name='top_block' style="position:absolute; top:15mm; left:10mm;">
```
Comment (MiquelRForgeFlow) [inline account_check_printing_report_sslm102/views/report_check.xml:9]: @hveficent could you check if this template needs a wkhtmltopdf 0.12.5 revamp like done in https://github.com/OCA/account-payment/pull/212?

File: account_check_printing_report_sslm102/views/report_check.xml [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        </t>
+    </template>
+
+    <template id="check_data">
+        <div>
+            <div name='amount_in_words' style="position:absolute; right:20mm; top:115mm;">
+                <strong><span t-esc="fill_stars(o.check_amount_in_words)"/></strong>
+            </div>
+            <div name='payment_date' style="position:absolute; right:60mm; top:128mm;">
+                <span t-esc="_format_date_to_partner_lang(o.payment_date, o.partner_id.id)"/>
```
Comment (HviorForgeFlow) [inline account_check_printing_report_sslm102/views/report_check.xml:38]: Functional review - Error found @mreficent 
```
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/addons/base/models/qweb.py", line 346, in _compiled_fn
    return compiled(self, append, new, options, log)
  File "<template>", line 1, in template_account_check_printing_report_sslm102_invoice_data_1326
TypeError: 'NoneType' object is not callable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/odoo/OCB-12.0/addons/web/controllers/main.py", line 1671, in report_download
    response = self.report_routes(reportname, docids=docids, converter=converter)
  File "/.repo_requirements/odoo/odoo/http.py", line 519, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/OCB-12.0/addons/web/controllers/main.py", line 1612, in report_routes
    pdf = report.with_context(context).render_qweb_pdf(docids, data=data)[0]
  File "/.repo_requirements/odoo/odoo/addons/base/models/ir_actions_report.py", line 711, in render_qweb_pdf
    html = self.with_context(context).render_qweb_html(res_ids, data=data)[0]
  File "/.repo_requirements/odoo/odoo/addons/base/models/ir_actions_report.py", line 751, in render_qweb_html
    return self.render_template(self.report_name, data), 'html'
  File "/.repo_requirements/odoo/odoo/addons/base/models/ir_actions_report.py", line 534, in render_template
    return view_obj.render_template(template, values)
  File "/.repo_requirements/odoo/odoo/addons/base/models/ir_ui_view.py", line 1304, in render_template
    return self.browse(self.get_view_id(template)).render(values, engine)
  File "/home/odoo/OCB-12.0/addons/web_editor/models/ir_ui_view.py", line 27, in render
    return super(IrUiView, self).render(values=values, engine=engine, minimal_qcontext=minimal_qcontext)
  File "/.repo_requirements/odoo/odoo/addons/base/models/ir_ui_view.py", line 1313, in render
    return self.env[engine].render(self.id, qcontext)
  File "/.repo_requirements/odoo/odoo/addons/base/models/ir_qweb.py", line 59, in render
    result = super(IrQWeb, self).render(id_or_xml_id, values=values, **context)
  File "/.repo_requirements/odoo/odoo/addons/base/models/qweb.py", line 274, in render
    self.compile(template, options)(self, body.append, values or {})
  File "/.repo_requirements/odoo/odoo/addons/base/models/qweb.py", line 348, in _compiled_fn
    raise e
  File "/.repo_requirements/odoo/odoo/addons/base/models/qweb.py", line 346, in _compiled_fn
    return compiled(self, append, new, options, log)
  File "<template>", line 1, in template_717_1313
  File "<template>", line 2, in body_call_content_1312
  File "<template>", line 5, in foreach_1311
  File "/.repo_requirements/odoo/odoo/addons/base/models/qweb.py", line 353, in _compiled_fn
    raise QWebException("Error to render compiling AST", e, path, node and etree.tostring(node[0], encoding='unicode'), name)
odoo.addons.base.models.qweb.QWebException: 'NoneType' object is not callable
Traceback (most recent call last):
  File "/.repo_requirements/odoo/odoo/addons/base/models/qweb.py", line 346, in _compiled_fn
    return compiled(self, append, new, options, log)
  File "<template>", line 1, in template_account_check_printing_report_sslm102_invoice_data_1326
TypeError: 'NoneType' object is not callable

Error to render compiling AST
TypeError: 'NoneType' object is not callable
Template: account_check_printing_report_sslm102.invoice_data
Path: /templates/t/div/strong/span[2]
Node: <span name="payment_date" style="padding-right:100mm;float:right;" t-esc="'Date: {}'.format(_format_date_to_partner_lang(o.payment_date, o.partner_id.id))"/>
```

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]

File: account_check_printing_report_sslm102/views/report_check.xml (hunk 2 above)
Comment (MiquelRForgeFlow) [reply account_check_printing_report_sslm102/views/report_check.xml:38]: solved

File: account_check_printing_report_sslm102/views/report_check.xml (hunk 1 above)
Comment (MiquelRForgeFlow) [reply account_check_printing_report_sslm102/views/report_check.xml:9]: @hveficent could you apply wkhtmltopdf 0.12.5 revamp, please?

Comment (HviorForgeFlow) [reply account_check_printing_report_sslm102/views/report_check.xml:9]: @mreficent for now it is ok! Let users decide how to apply styling on demand :smile:

**Review by HviorForgeFlow — APPROVED** [review]

**Review by MiquelRForgeFlow — COMMENTED** [review]
:+1:

**Review by AaronHForgeFlow — APPROVED** [review]
Tested. LGTM :+1:

**Comment by AaronHForgeFlow** [comment]
Ready to merge

**Pull request merged**